docs/game/unity/how-tos/update-events.md

```diff
@@ -0,0 +1,231 @@
+
+Smart contracts emit `Events` inside methods. An `Event` can be likened to a record in a database. This record can be indexed and is searchable. `Events` are also linked to the block during which they were emitted.
+
+*Indexing*
+
+The EVM restricts the number of indexed fields to three, however the `Event` itself can have any number of fields. 
+
+```solidity
+emit Transfer(from, to, amount);
+```
+
+So now we understand what `Events` are.
+
+## How to prepare a DTO for request events
+
+1) First we need to create a class extended from `EventDTOBase`. Add the same fields as for the `Event` itself. 
+
+:::tip Naming Class fields
+
+We recommend naming your class fields with the same names as the props of the `Event` in the contract
+
+:::
+
+2) Add the attribute `Event` to the class with the same name as the `Event` in the contract e.g. `TransferEventDTO`.
+
+3) Add the attribute `Parameter` for every field in the class and pass the types of `Event` props e.g.
+specify the type and name of the `Even`t prop, the order of the prop in the `event` and whether the prop is indexed or not as a boolean. 
+
+```c
+public ParameterAttribute(string type, string name, int order, bool indexed = false)
+```
+
+:::tip
+
+Make sure you use compatible field types for types of Event props
+
+:::
+
+
+```c
+
+[Event("Transfer")]
+public class TransferEventDTO : EventDTOBase
+{
+	[Parameter("address", "from", 1, true)]
+	public string From { get; set; }
+
+	[Parameter("address", "to", 2, true)]
+	public string To { get; set; }
+
+	[Parameter("uint256", "value", 3, false)]
+	public BigInteger Value { get; set; }
+}
+```
+
+
+Pay attention to the use of compatible field types for the type of `Event` props.
+
+|Type in C#  |  Type in Solidity |
+|------------|-------------------|
+|string      |   address         |
+|BigInteger  | uint8 to uint256  |
+|byte[]      |bytes, bytes1 to bytes32|
+|bool        |bool               |
+|string      |string             |
+
+## Understanding requests for `events`
+
+So before we continue, let's see what requests to an Ethereum node look like and what the limitation on requests are:
+
+
+* `fromBlock` - This is the block from which to start searching from (it can be a specific number as well as key words like latest, earliest, pending );
+
+* `toBlock` - This is the block that we search up to. (The value can be the same as `fromBlock`);
+
+* `topic1` - values that are relevant to the first indexing of event props e.g. `from address`.
+
+* `topic2` - values that are relevant to the second indexing of event props. e.g. `to address`.
+
+* `topic3` - values that are relevant to the third indexing of event props e.g. `value`.
+
+:::tip What are Topics?
+
+Topics are an array of values that describe what's going on in an event. 
+
+The order is important, if you want to leave topics out use null, e.g. `[null, '0x00...']`. 
+
+You can pass another array for each topic with options for that topic e.g. [null, ['option1', 'option2']]
```

Comment:
I think we need to mention here that all values of array will be logically connected using OR operator (example 'option1' OR 'option2' OR 'option3')

Reply:
Done.